src/dune_rules/exe_rules.ml

```diff
@@ -160,25 +160,28 @@ let executables_rules ~sctx ~dir ~expander ~dir_contents ~scope ~compile_info
       and+ ctypes_cclib_flags =
         Ctypes_rules.ctypes_cclib_flags sctx ~expander ~buildable:exes.buildable
       in
-      Command.Args.S
-        [ As flags
-        ; S
-            (let ext_lib = ctx.lib_config.ext_lib in
-             let foreign_archives =
-               exes.buildable.foreign_archives |> List.map ~f:snd
-             in
-             (* XXX: don't these need the msvc hack being done in lib_rules? *)
-             (* XXX: also the Command.quote_args being done in lib_rules? *)
-             List.map foreign_archives ~f:(fun archive ->
-                 let lib =
-                   Foreign.Archive.lib_file ~archive ~dir ~ext_lib
-                     ~mode:Mode.Select.All
-                 in
-                 Command.Args.S [ A "-cclib"; Dep (Path.build lib) ]))
-          (* XXX: don't these need the msvc hack being done in lib_rules? *)
-          (* XXX: also the Command.quote_args being done in lib_rules? *)
-        ; As (List.concat_map ctypes_cclib_flags ~f:(fun f -> [ "-cclib"; f ]))
-        ]
+      let has_linkall = List.mem ~equal:String.equal flags "-linkall" in
+      ( `Linkall has_linkall
```

Comment:
@rgrinberg, I think this call site is the reason we don't have a `string list` for `link_args`..

Reply:
I see. I suppose we can easily convert it to a list here as well, but that would be counter productive.

Reply:
This is no longer relevant